chartfx-chart/src/test/java/de/gsi/chart/axes/spi/AbstractAxisTests.java

```diff
@@ -0,0 +1,293 @@
+package de.gsi.chart.axes.spi;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javafx.scene.Scene;
+import javafx.scene.canvas.Canvas;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.layout.Pane;
+import javafx.stage.Stage;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.function.ThrowingSupplier;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.testfx.framework.junit5.ApplicationExtension;
+import org.testfx.framework.junit5.Start;
+
+import de.gsi.chart.axes.AxisLabelFormatter;
+import de.gsi.chart.axes.AxisTransform;
+import de.gsi.chart.axes.LogAxisType;
+import de.gsi.chart.axes.spi.transforms.DefaultAxisTransform;
+import de.gsi.chart.legend.spi.DefaultLegend;
+import de.gsi.chart.ui.geometry.Side;
+import de.gsi.chart.ui.utils.JavaFXInterceptorUtils;
+import de.gsi.chart.utils.FXUtils;
+
+@ExtendWith(ApplicationExtension.class)
+@ExtendWith(JavaFXInterceptorUtils.SelectiveJavaFxInterceptor.class)
+class AbstractAxisTests {
+    private static final int WIDTH = 300;
+    private static final int HEIGHT = 200;
+    private Pane pane;
```

Comment:
Codacy found an issue: [Perhaps 'pane' could be replaced by a local variable.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6223548)